packages/controller/src/lib/setup.ts

```diff
@@ -470,42 +483,47 @@ function initYargs() {
         .command(['version [<adapter>]', 'v [<adapter>]'], 'Show version of js-controller or specified adapter')
         .wrap(null);
 
-    return yargs;
+    return _yargs;
 }
 
 /**
  * Show yargs help, if processCommand is used as import, yargs won't be initialized
- * @param {object?} _yargs - yargs instance
+ * @param _yargs - yargs instance
  */
-function showHelp(_yargs) {
+function showHelp(_yargs?: yargs.Argv) {
     if (_yargs) {
         _yargs.showHelp();
-    } else if (yargs) {
+    } else {
         yargs.showHelp();
     }
 }
 
-let Objects; // constructor
-let objects; // instance
-let States; // constructor
-let states; // instance
+let Objects: typeof ObjectsInRedisClient | null; // constructor
+let objects: ObjectsInRedisClient | null; // instance
+let States: typeof StateRedisClient | null; // constructor
+let states: StateRedisClient | null; // instance
 
 /**
  * Process the given CLI command
  *
- * @param {string|number} command - command to execute
- * @param {any[]} args - arguments passed to yargs
- * @param {object} params - object with parsed params by yargs, e. g. --force is params.force
- * @param {(exitCode?: number) => void} callback
+ * @param command - command to execute
+ * @param args - arguments passed to yargs
+ * @param params - object with parsed params by yargs, e. g. --force is params.force
+ * @param callback
  */
-async function processCommand(command, args, params, callback) {
+async function processCommand(
```

Comment:
Looks like this may need overloads?

Reply:
no really, we only have one call in this file. I removed the unnecessary sanity check juggle